test/unit/reducers/projects.js

```diff
@@ -87,7 +88,11 @@ test('changeCurrentProject', (t) => {
     reducer,
     initProjects({1: true, 2: false}),
     partial(changeCurrentProject, '2'),
-    initProjects({1: true, 2: false}),
+    initProjects({1: true, 2: false}).
+      update(
+        '2',
+        editedProject => editedProject.set('isArchived', false),
```

Comment:
So I believe the idea here is to test the behavior of projects getting unarchived when you switch to them, BUT:

1. It’s not immediately clear from looking at this that that is the behavior under test
2. I’m not even sure that that behavior _is_ being tested? It seems like in the initial state both projects are not archived, so the assertion here would pass even if the reducer side of things had not been implemented (unless I am missing something)

I think the immediate fix for both issues would be a separate test that explicitly sets the initial state of the project to archived, and then explicitly expects the output state to be un-archived.

More broadly this gets right to the heart of my critique of the way our reducer tests are structured (my fault!)—because we make assertions about the entire output state, rather than making discrete assertions about the things in the state we actually care about for the given test, it’s quite difficult to divine the intent of the test a lot of the time (and many tests are implicitly testing several behaviors, which is undesirable for a unit test)